### ISSUE TITLE
Add Github action

### ISSUE THREAD

**Issue by FFY00** [pypa/build#781]
Similar to https://github.com/pypa/gh-action-pypi-publish but to build packages.

**Comment by henryiii** [comment]
I think this would basically be https://github.com/hynek/build-and-inspect-python-package except without the validation? Edit: also no auto archive uploads.

**Comment by FFY00** [comment]
Yes, I wasn't aware of that, maybe it doesn't make much sense then though.

**Comment by henryiii** [comment]
Build uses it. :) It also uses uv and our `--installer=uv` option, so it's really fast. Not sure we could do better. If we could provide an action that was flexible enough that action could use, though, I think that might be worth while.

(Actually, I use it in so many of my repos that I spent 10 minutes looking for an example of someone learning recently to find one that didn't use it)

**Comment by FFY00** [comment]
Okay, today I ran into a use-case where that action is a bit undesirable. I want to build a wheel as an intermediary step, and hynek's action always uploads the artifact, which is not needed.

**Comment by gaborbernat** [comment]
The requested GitHub Actions documentation has been added in #988. The docs now include a comprehensive CI/CD guide at https://build.pypa.io/en/stable/how-to/ci-cd.html which recommends hynek/build-and-inspect-python-package as the primary GitHub Action for building packages.

This action is already used by the build project itself in its CD workflow. The documentation covers:
- Using hynek's build-and-inspect-python-package action (recommended)
- Manual builds with isolation
- Fast builds with uv
- Matrix builds across Python versions
- Platform-specific builds with cibuildwheel
- Publishing to PyPI

Closing this issue as the documentation is now available.
